src/core/public/theme/core_theme_provider.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FC, useMemo } from 'react';
+import { Observable } from 'rxjs';
+import useObservable from 'react-use/lib/useObservable';
+import { EuiThemeProvider } from '@elastic/eui';
+import { CoreTheme } from './types';
+import { convertCoreTheme } from './convert_core_theme';
+
+const defaultTheme: CoreTheme = {
+  darkMode: false,
+};
+
+interface CoreThemeProviderProps {
+  theme$: Observable<CoreTheme>;
+}
+
+/**
+ * Wrapper around `EuiThemeProvider` converting (and exposing) core's theme to EUI theme.
+ * @internal Only meant to be used within core for internal usages of EUI/React
+ */
+export const CoreThemeProvider: FC<CoreThemeProviderProps> = ({ theme$, children }) => {
+  const coreTheme = useObservable(theme$, defaultTheme);
+  const euiTheme = useMemo(() => convertCoreTheme(coreTheme), [coreTheme]);
+  return (
+    <EuiThemeProvider colorMode={euiTheme.colorMode} theme={euiTheme.euiThemeSystem}>
```

Comment:
`euiThemeSystem` is currently always `undefined`, as we only have `colorMode` converted from Kibana's `darkMode`, but this is ready for when/if we want to allow theme customization.

Reply:
Question for @cchaos: This is using the (soon-to-be) legacy theme until https://github.com/elastic/eui/pull/5121 merges. Should we configure to use Amsterdam in in the meantime?

Reply:
Yes it should definitely be rendering the Amsterdam theme which is the same as Kibana's v8 theme. We may want to create a map of Kibana's theme naming to EUI's in order to present the right selected theme. Luckily right now we're only dealing with v8 and not the old v7 theme too.

Reply:
Ok. I can look into this.

Reply:
https://github.com/elastic/eui/pull/5374 has additional types requested by @pgayvallet and enables using Amsterdam by default. It'll be part of this week's EUI release and hopefully in Kibana next week.